### ISSUE TITLE
Supporting HSTS and HSTS preloading

### ISSUE THREAD

**Issue by munwr** [gatsbyjs/gatsby#6451]
## Summary

As we are already serving websites in HTTPS, it is recommended to support HSTS (HTTP Strict Transport Security). [Secure your site with HTTPS](https://support.google.com/webmasters/answer/6073543?hl=en)

HSTS tells the browser to request HTTPS pages automatically, even if the user enters `http` in the browser location bar. It also tells Google to serve secure URLs in the search results. All this minimizes the risk of serving unsecured content to your users.

HSTS preload results — https://hstspreload.org/?domain=gatsbyjs.org

<img width="964" alt="screen shot 2018-07-14 at 3 14 42 pm" src="https://user-images.githubusercontent.com/6573105/42723282-c4713604-8778-11e8-86c2-b92fa1680c94.png">

### Basic example
This would involve passing the following header.
` {
      resolve: `gatsby-plugin-netlify`,
      options: {
        headers: {
          "/*": ["Strict-Transport-Security: max-age=63072000; includeSubDomains; preload",]
        },
      }
    },`

and redirecting our site to `gatsby.org` instead of `www.gatsby.org` along with a 301 redirect for `www.gatsby.org` to `gatsby.org`. So, search engines follow.

----
Please, provide your suggestions on this. @KyleAMathews @jlengstorf @fk

**Comment by jlengstorf** [comment]
Nice! We have a redirect in place already, but this seems like a nice addition.

Unless @KyleAMathews or @pieh has concerns, this seems like a good thing to PR.

**Comment by munwr** [comment]
@KyleAMathews If we want to stay at `www`, then there is no problem :)

But if we want HSTS preload then we need to push to bare gatsbyjs.org. As we can see its one of the requirements in HSTS preload results — https://hstspreload.org/?domain=gatsbyjs.org

Netlify does provide an option to do this. After visiting our site domain settings, we have three options
1. Netlify subdomain URL
2. Domain URL (if we set this to the primary domain, it would redirect to bare domain.tld)
3. Domain URL with WWW

I did the same for my domain
<img width="1239" alt="screen shot 2018-07-15 at 3 26 45 pm" src="https://user-images.githubusercontent.com/6573105/42732700-8cdc8222-8843-11e8-905e-1f8a47a516be.png">

and submitted the same to HSTS preload. https://hstspreload.org/?domain=munwr.com
